packages/flutter/test/widgets/single_child_scroll_view_test.dart

```diff
@@ -581,8 +581,8 @@ void main() {
                   children: List<Widget>.generate(10, (int y) {
                     return Row(
                       children: children[y] = List<Widget>.generate(10, (int x) {
-                        return Container(
-                          height: 100.0,
+                        return SizedBox(
+                          height: nonconst(100.0),
```

Comment:
Not clear why this transformation is needed. Doesn't seem to improve the clarity over the original

Reply:
Without `nonconst` `SizedBox` should be made const. Doing that breaks the test on L603 with the following error:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following StateError was thrown running a test:
Bad state: Too many elements

When the exception was thrown, this was the stack:
#0      Iterable.single (dart:core/iterable.dart:556:24)
#1      WidgetController.renderObject (package:flutter_test/src/controller.dart:207:30)
#2      main.<anonymous closure> (file:///home/a14n/flutter/flutter/packages/flutter/test/widgets/single_child_scroll_view_test.dart:603:12)
<asynchronous suspension>
#3      main.<anonymous closure> (file:///home/a14n/flutter/flutter/packages/flutter/test/widgets/single_child_scroll_view_test.dart)
#4      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:140:29)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:703:19)
<asynchronous suspension>
#9      TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:683:14)
#10     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1083:24)
#16     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1080:15)
#17     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:133:24)
#18     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:172:27)
<asynchronous suspension>
#19     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#20     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:246:15)
#25     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:243:5)
#26     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:170:33)
#31     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:13)
#32     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:400:30)
(elided 31 frames from class _FakeAsync, class _RawReceivePortImpl, class _Timer, dart:async, dart:async-patch, and package:stack_trace)

The test description was:
  Nested SingleChildScrollView showOnScreen
════════════════════════════════════════════════════════════════════════════════════════════════════
Test failed. See exception logs above.
The test description was: Nested SingleChildScrollView showOnScreen

✖ Nested SingleChildScrollView showOnScreen
```

> Doesn't seem to improve the clarity of the original code and I'm sure we don't care about the performance difference :-).

There's an upcoming lint to detect those empty Containers. So we should change this code.

Reply:
The test needs a different Container (now SizedBox) in every cell of its 10x10 grid, otherwise it's not possible to distinguish individual children. I think it would be better to either use `nonconst SizedBox(...)` with a comment that explains what's going on, or add an ignore analyzer rule comment (once the new rule lands) with the same comment.

Reply:
`nonconst` was designed to address this case and avoid an `// ignore: prefer_const_constructors`. Looking at other places where `nonconst` is used, it looks really similar to this case.

Reply:
OK. It's the first time I've ever seen it, so if there's a convention, I'll bow to it. Please add a comment that explains what's up, for future developers who (like me) haven't run into this idiom before.

Reply:
Comment added